docs/getting-started/creating-function.md

```diff
@@ -0,0 +1,26 @@
+# Creating a Knative Function
+
+Knative Functions provide a programming model that leverages the Knative
+Serving and eventing APIs to provide a quick and easy way for developers
+to deploy their first Knative applications. Function lifecycle management
+includes creating, building, and deploying a function. Optionally, you can
+also test a deployed function by invoking it locally or on a cluster.
+You can do all of these operations using the `func` command line interface.
+
+In this tutorial, you will create, build and deploy a Knative Function
+that responds to CloudEvents emitted by a Ping Source. Supported languages
+for Knative Functions include Go, JavaScript, TypeScript, Java and Rust.
+In this tutorial, you will use Go.
+
+=== "func"
+
+    Create the function by running the command:
+
+    ```bash
+    func create -l go -t cloudevents hello
```

Comment:
I would use `kn func` instead of func

Reply:
The installation of func plugin for `kn` are missing in the tutorial, maybe they should be after install kn, go ahead and install the func plugin.

Reply:
> I would use `kn func` instead of func

I am curious why. To me that seems to add additional friction. The user is required to install `kn`, then install `func`, and make sure that `func` is installed in the right way/place in order for it to work as a plugin correctly.

Reply:
+1 for `kn func`, mostly because it's just more consistent from a UX perspective. Adding `func` as another CLI in its own right rather than it being a plugin of `kn` to me is more confusing and will lead to more questions, e.g. is `kn` required at all then or should users just use `func` and YAML / `kubectl`? I think from a user perspective, even though there might be an extra step in the setup, `kn func` is easier to remember.

Reply:
I just don't think it's nearly as easy to set up `kn func` as it is to simply use `func`. To set up `kn func` the user has two downloads, then must place each of them in the correct location on their system. Additionally, it's not always clear where `kn` picks up the plugins from. Let's say a user has `func` in both `~/bin/` and `/opt/local/bin` and they are two different versions. Which one will `kn` use? It's not obvious. Whereas, with simply `func` the user has to type `brew install func` and it's done - or download only `func` from an upstream release - in either case, there is much less opportunity for user error and confusing messages.

Reply:
I would use `func` too.. adding `kn` and treating `func` just as a plugin feels unnatural to me, if you are only working with functions.

Reply:
The simplest path is to stick with `func` here, and call out to a plugin installation guide as necessary.

I understand the desire to explain and introduce everything we have, from the bottom up, but every extra command that we expect a user to run; every additional concept; will have some nontrivial reduction in the number of people who make it through the learning curve gauntlet.